doc build fails on Python 3.8 + MacOS

<!--- Verify first that your improvement is not already reported on GitHub -->
<!--- Also test if the latest release and devel branch are affected too -->
<!--- Complete *all* sections as described, this form is processed automatically -->

##### SUMMARY
<!--- Explain the problem briefly below, add suggestions to wording or structure -->

Documentation doesn't currently build on MacOS with Python3.8.

This is due to https://github.com/sphinx-doc/sphinx/issues/6803 which is not fixed in the sphinx version used by ansible documentation build system (it's frozen to 2.1.2)


<!--- HINT: Did you know the documentation has an "Edit on GitHub" link on every page ? -->

##### ISSUE TYPE
- Documentation Report

##### COMPONENT NAME
<!--- Write the short name of the rst file, module, plugin, task or feature below, use your best guess if unsure -->

[docs/docsite/requirements.txt](docs/docsite/requirements.txt)

##### ANSIBLE VERSION
<!--- Paste verbatim output from "ansible --version" between quotes -->
```paste below
ansible 2.9.11
  config file = None
  configured module search path = ['/Users/waldekm/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/local/Cellar/ansible/2.9.11/libexec/lib/python3.8/site-packages/ansible
  executable location = /usr/local/bin/ansible
  python version = 3.8.5 (default, Jul 21 2020, 10:48:26) [Clang 11.0.3 (clang-1103.0.32.62)]
```

##### CONFIGURATION
<!--- Paste verbatim output from "ansible-config dump --only-changed" between quotes -->
```paste below
# empty output
```

##### OS / ENVIRONMENT
<!--- Provide all relevant information below, e.g. OS version, browser, etc. -->

```
$ uname -rs
Darwin 19.6.0

$ python --version
Python 3.8.5
```

##### ADDITIONAL INFORMATION
<!--- Describe how this improves the documentation, e.g. before/after situation or screenshots -->

Error log

```
$ make webdocs
(cd docs/docsite/; CPUS=8 /Applications/Xcode.app/Contents/Developer/usr/bin/make docs)
../../hacking/build-ansible.py collection-meta --template-file=../templates/collections_galaxy_meta.rst.j2 --output-dir=rst/dev_guide/ ../../lib/ansible/galaxy/data/collections_galaxy_meta.yml
../../hacking/build-ansible.py document-config --template-file=../templates/config.rst.j2 --output-dir=rst/reference_appendices/ ../../lib/ansible/config/base.yml
mkdir -p rst/cli
../../hacking/build-ansible.py generate-man --template-file=../templates/cli_rst.j2 --output-dir=rst/cli/ --output-format rst ../../lib/ansible/cli/*.py
[WARNING]: You are running the development version of Ansible. You should only run Ansible from "devel" if you are modifying the Ansible engine, or
trying out features under development. This is a rapidly changing source of code and can become unstable at any point.
../../hacking/build-ansible.py document-keywords --template-dir=../templates --output-dir=rst/reference_appendices/ ./keyword_desc.yml
if expr "2.11.0.dev0" : '.*[.]dev[0-9]\{1,\}$' &> /dev/null; then \
		../../hacking/build-ansible.py docs-build base -o rst ;\
	else \
		../../hacking/build-ansible.py docs-build full -o rst ;\
	fi
ERROR:antsibull:func=write_rst:mod=antsibull.write_docs:nonfatal_errors=['1 validation error for PluginDocSchema\ndoc -> options -> cache -> deprecated -> removed_from_collection\n  field required (type=value_error.missing)']:plugin_name=ansible.builtin.script:plugin_type=inventory|ansible.builtin.script did not return correct DOCUMENTATION.  An error page will be generated.
ERROR:antsibull:func=write_rst:mod=antsibull.write_docs:nonfatal_errors=['18 validation errors for PluginDocSchema\ndoc -> options -> ca_path -> ini -> 0 -> key\n  field required (type=value_error.missing)\ndoc -> options -> ca_path -> ini -> 1 -> section\n  field required (type=value_error.missing)\ndoc -> options -> follow_redirects -> ini -> 0 -> key\n  field required (type=value_error.missing)\ndoc -> options -> follow_redirects -> ini -> 1 -> section\n  field required (type=value_error.missing)\ndoc -> options -> force -> ini -> 0 -> key\n  field required (type=value_error.missing)\ndoc -> options -> force -> ini -> 1 -> section\n  field required (type=value_error.missing)\ndoc -> options -> force_basic_auth -> ini -> 0 -> key\n  field required (type=value_error.missing)\ndoc -> options -> force_basic_auth -> ini -> 1 -> section\n  field required (type=value_error.missing)\ndoc -> options -> http_agent -> ini -> 0 -> key\n  field required (type=value_error.missing)\ndoc -> options -> http_agent -> ini -> 1 -> section\n  field required (type=value_error.missing)\ndoc -> options -> timeout -> ini -> 0 -> key\n  field required (type=value_error.missing)\ndoc -> options -> timeout -> ini -> 1 -> section\n  field required (type=value_error.missing)\ndoc -> options -> unix_socket -> ini -> 0 -> key\n  field required (type=value_error.missing)\ndoc -> options -> unix_socket -> ini -> 1 -> section\n  field required (type=value_error.missing)\ndoc -> options -> unredirected_headers -> ini -> 0 -> key\n  field required (type=value_error.missing)\ndoc -> options -> unredirected_headers -> ini -> 1 -> section\n  field required (type=value_error.missing)\ndoc -> options -> use_gssapi -> ini -> 0 -> key\n  field required (type=value_error.missing)\ndoc -> options -> use_gssapi -> ini -> 1 -> section\n  field required (type=value_error.missing)']:plugin_name=ansible.builtin.url:plugin_type=lookup|ansible.builtin.url did not return correct DOCUMENTATION.  An error page will be generated.
../bin/testing_formatter.sh
../bin/testing_formatter.sh: line 38: ../docsite/rst/dev_guide/testing/sanity/index.rst: No such file or directory
CPUS=8 /Applications/Xcode.app/Contents/Developer/usr/bin/make -f Makefile.sphinx html
sphinx-build -M html "rst" "_build" -j 8 -n -w rst_warnings
Running Sphinx v2.1.2
making output directory... done
loading intersphinx inventory from https://docs.python.org/2/objects.inv...
loading intersphinx inventory from https://docs.python.org/3/objects.inv...
loading intersphinx inventory from http://jinja.palletsprojects.com/objects.inv...
intersphinx inventory has moved: http://jinja.palletsprojects.com/objects.inv -> https://jinja.palletsprojects.com/en/2.11.x/objects.inv
loading intersphinx inventory from https://docs.ansible.com/ansible/2.10/objects.inv...
loading intersphinx inventory from https://docs.ansible.com/ansible/2.9/objects.inv...
loading intersphinx inventory from https://docs.ansible.com/ansible/2.8/objects.inv...
loading intersphinx inventory from https://docs.ansible.com/ansible/2.7/objects.inv...
loading intersphinx inventory from https://docs.ansible.com/ansible/2.6/objects.inv...
loading intersphinx inventory from https://docs.ansible.com/ansible/2.5/objects.inv...
building [mo]: targets for 0 po files that are out of date
building [html]: targets for 463 source files that are out of date
updating environment: 463 added, 0 changed, 0 removed
reading sources... [  5%] 404 .. collections/ansible/builtin/config_lookup
Exception occurred:
  File "/usr/local/opt/python@3.8/Frameworks/Python.framework/Versions/3.8/lib/python3.8/multiprocessing/reduction.py", line 60, in dump
    ForkingPickler(file, protocol).dump(obj)
AttributeError: Can't pickle local object 'Builder._read_parallel.<locals>.merge'
The full traceback has been saved in /var/folders/f1/svj655pj0mj11j240_yh47840000gn/T/sphinx-err-w96ydjs6.log, if you want to report the issue to the developers.
Please also report this if it was a user error, so that a better error message can be provided next time.
A bug report can be filed in the tracker at <https://github.com/sphinx-doc/sphinx/issues>. Thanks!
make[2]: *** [html] Error 2
make[1]: *** [htmldocs] Error 2
make: *** [webdocs] Error 2
```

<!--- HINT: You can paste gist.github.com links for larger files -->

--- a/docs/docsite/known_good_reqs.txt
+++ b/docs/docsite/known_good_reqs.txt
@@ -1,0 +1,14 @@
+# pip packages required to build docsite
+# tested June 9 2021
+jinja2==3.0.1
+PyYAML==5.4.1
+rstcheck==3.3.1
+sphinx==4.0.2
+sphinx-notfound-page==0.7.1 # must be >= 0.6
+sphinx-intl==2.0.1
+sphinx_ansible_theme==0.6.0
+resolvelib==0.5.4
+Pygments==2.9.0 # must be >= 2.4.0
+straight.plugin==1.5.0 # Needed for hacking/build-ansible.py which is the backend build script
+antsibull==0.33.0 # must be >= 0.25.0
+docutils==0.16 # pinned globally until the problem with unordered lists is fixed, see https://github.com/readthedocs/sphinx_rtd_theme/issues/1115

--- a/docs/docsite/known_good_reqs.txt
+++ b/docs/docsite/known_good_reqs.txt
@@ -1,14 +1,17 @@
 # pip packages required to build docsite
 # tested June 9 2021
+
+antsibull==0.33.0
+docutils==0.16
+# check unordered lists when testing more recent docutils versions
+# see https://github.com/readthedocs/sphinx_rtd_theme/issues/1115
 jinja2==3.0.1
+Pygments==2.9.0
 PyYAML==5.4.1
+resolvelib==0.5.4
 rstcheck==3.3.1
 sphinx==4.0.2
 sphinx-notfound-page==0.7.1 # must be >= 0.6
 sphinx-intl==2.0.1
 sphinx_ansible_theme===0.6.0
-resolvelib==0.5.4
-Pygments==2.9.0 # must be >= 2.4.0
 straight.plugin==1.5.0 # Needed for hacking/build-ansible.py which is the backend build script
-antsibull==0.33.0 # must be >= 0.25.0
-docutils==0.16 # pinned globally until the problem with unordered lists is fixed, see https://github.com/readthedocs/sphinx_rtd_theme/issues/1115

--- a/docs/docsite/known_good_reqs.txt
+++ b/docs/docsite/known_good_reqs.txt
@@ -6,7 +6,7 @@ rstcheck==3.3.1
 sphinx==4.0.2
 sphinx-notfound-page==0.7.1 # must be >= 0.6
 sphinx-intl==2.0.1
-sphinx_ansible_theme==0.6.0
+sphinx_ansible_theme===0.6.0
 resolvelib==0.5.4
 Pygments==2.9.0 # must be >= 2.4.0
 straight.plugin==1.5.0 # Needed for hacking/build-ansible.py which is the backend build script

--- a/docs/docsite/requirements.txt
+++ b/docs/docsite/requirements.txt
@@ -1,13 +1,16 @@
-#pip packages required to build docsite
+# pip packages required to build docsite
+# these requirements are as loosely defined as possible
+# if you want known good versions of these dependencies
+# use known_good_reqs.txt instead 
 jinja2
 PyYAML
 rstcheck
 sphinx
 sphinx-notfound-page >= 0.6
 sphinx-intl
-sphinx_ansible_theme === 0.6.0
+sphinx_ansible_theme == 0.6.0
 resolvelib
 Pygments >= 2.4.0
 straight.plugin # Needed for hacking/build-ansible.py which is the backend build script
 antsibull >= 0.25.0
-docutils==0.16 # pin for now until sphinx_rtd_theme is compatible with 0.17 or later
+docutils == 0.16 # pin for now until sphinx_rtd_theme is compatible with 0.17 or later

--- a/docs/docsite/requirements.txt
+++ b/docs/docsite/requirements.txt
@@ -2,7 +2,12 @@
 # these requirements are as loosely defined as possible
 # if you want known good versions of these dependencies
 # use known_good_reqs.txt instead 
+
+antsibull >= 0.25.0
+docutils == 0.16 # pin for now until the problem with unordered lists is fixed
+# see https://github.com/readthedocs/sphinx_rtd_theme/issues/1115jinja2==3.0.1
 jinja2
+Pygments >= 2.4.0
 PyYAML
 rstcheck
 sphinx
@@ -10,7 +15,4 @@ sphinx-notfound-page >= 0.6
 sphinx-intl
 sphinx_ansible_theme === 0.6.0
 resolvelib
-Pygments >= 2.4.0
 straight.plugin # Needed for hacking/build-ansible.py which is the backend build script
-antsibull >= 0.25.0
-docutils == 0.16 # pin for now until sphinx_rtd_theme is compatible with 0.17 or later

--- a/docs/docsite/requirements.txt
+++ b/docs/docsite/requirements.txt
@@ -8,7 +8,7 @@ rstcheck
 sphinx
 sphinx-notfound-page >= 0.6
 sphinx-intl
-sphinx_ansible_theme == 0.6.0
+sphinx_ansible_theme === 0.6.0
 resolvelib
 Pygments >= 2.4.0
 straight.plugin # Needed for hacking/build-ansible.py which is the backend build script

--- a/docs/docsite/requirements.txt
+++ b/docs/docsite/requirements.txt
@@ -2,7 +2,7 @@
 jinja2
 PyYAML
 rstcheck
-sphinx==2.1.2
+sphinx
 sphinx-notfound-page >= 0.6
 sphinx-intl
 sphinx_ansible_theme === 0.6.0

--- a/docs/docsite/rst/community/documentation_contributions.rst
+++ b/docs/docsite/rst/community/documentation_contributions.rst
@@ -74,22 +74,7 @@ Setting up your environment to build documentation locally
 
 To build documentation locally, ensure you have a working :ref:`development environment <environment_setup>`.
 
-To work with documentation on your local machine, you need to have python-3.5 or greater and the following packages installed:
-
-    - ``gcc``
-    - ``jinja2``
-    - ``libyaml``
-    - ``make``
-    - ``Pygments``
-    - ``pyparsing``
-    - ``PyYAML``
-    - ``rstcheck``
-    - ``six``
-    - ``sphinx``
-    - ``sphinx-notfound-page``
-    - ``straight.plugin``
-
-These required packages are listed in our :file:`requirements.txt` files to make installation easier:
+To work with documentation on your local machine, you need to have python-3.5 or greater and install the `Ansible dependencies <https://github.com/ansible/ansible/blob/devel/requirements.txt>`_ and `documentation dependencies<https://github.com/ansible/ansible/blob/devel/docs/docsite/requirements.txt>`_, which are listed in two :file:`requirements.txt` files to make installation easier:
 
 .. code-block:: bash
 
@@ -107,6 +92,12 @@ You can drop ``--user`` if you have set up a virtual environment (venv/virtenv).
 
 .. note::
 
+    You may need to install these general pre-requisites separately on some systems:
+    - ``gcc``
+    - ``libyaml``
+    - ``make``
+    - ``pyparsing``
+    - ``six``
     On macOS with Xcode, you may need to install ``six`` and ``pyparsing`` with ``--ignore-installed`` to get versions that work with ``sphinx``.
 
 .. note::

--- a/docs/docsite/rst/community/documentation_contributions.rst
+++ b/docs/docsite/rst/community/documentation_contributions.rst
@@ -74,7 +74,10 @@ Setting up your environment to build documentation locally
 
 To build documentation locally, ensure you have a working :ref:`development environment <environment_setup>`.
 
-To work with documentation on your local machine, you need to have python-3.5 or greater and install the `Ansible dependencies <https://github.com/ansible/ansible/blob/devel/requirements.txt>`_ and `documentation dependencies<https://github.com/ansible/ansible/blob/devel/docs/docsite/requirements.txt>`_, which are listed in two :file:`requirements.txt` files to make installation easier:
+To work with documentation on your local machine, you need to have python-3.5 or greater and install the `Ansible dependencies`_ and `documentation dependencies`_, which are listed in two :file:`requirements.txt` files to make installation easier:
+
+.. _Ansible dependencies: https://github.com/ansible/ansible/blob/devel/requirements.txt
+.. _documentation dependencies: https://github.com/ansible/ansible/blob/devel/docs/docsite/requirements.txt
 
 .. code-block:: bash
 

--- a/docs/docsite/rst/community/documentation_contributions.rst
+++ b/docs/docsite/rst/community/documentation_contributions.rst
@@ -74,8 +74,7 @@ Setting up your environment to build documentation locally
 
 To build documentation locally, ensure you have a working :ref:`development environment <environment_setup>`.
 
-To work with documentation on your local machine, you need to have python-3.5 or greater and the
-following packages installed:
+To work with documentation on your local machine, you need to have python-3.5 or greater and the following packages installed:
 
     - ``gcc``
     - ``jinja2``
@@ -90,14 +89,21 @@ following packages installed:
     - ``sphinx-notfound-page``
     - ``straight.plugin``
 
-These required packages are listed in two :file:`requirements.txt` files to make installation easier:
+These required packages are listed in our :file:`requirements.txt` files to make installation easier:
 
 .. code-block:: bash
 
     pip install --user -r requirements.txt
     pip install --user -r docs/docsite/requirements.txt
 
-You can drop ``--user`` if you have set up a virtual environment (venv/virtenv).
+The :file:`docs/docsite/requirements.txt` file allows a wide range of versions and may install new releases of required packages. New releases of these packages may cause problems with the Ansible docs build. If you want to install tested versions of these dependencies, use :file:`docs/docsite/known_good_reqs.txt` instead:
+
+.. code-block:: bash
+
+    pip install --user -r requirements.txt
+    pip install --user -r docs/docsite/known_good_reqs.txt
+
+You can drop ``--user`` if you have set up a virtual environment (venv/virtenv). 
 
 .. note::
 

